### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by zerok**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,11 +2,7 @@
 
 var Hapi = require('hapi');
 var Version = require('./version');
-
-
-// Declare internals
-
-var internals = {};
```
Comment (AdriVanHoudt) [inline lib/index.js]: why did you remove internals?

File: lib/private.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+var Basic = require('hapi-auth-basic');
+var Util = require('util');
+var Hoek = require('hoek');
+
+
+// Declare internals
+
+var internals = {};
+internals.users = require('./users.json');
```
Comment (AdriVanHoudt) [inline lib/private.js]: this doesn't need to be in internals, better to just declare like hoek for example

**Comment by AdriVanHoudt** [comment]
general linting issues
```js
Linting results:
	lib/private.js:
		Line 23: no-else-return - Unexpected 'else' after 'return'.
		Line 31: space-before-function-paren - Missing space before function parentheses.
	test/private.js:
		Line 14: hapi/hapi-scope-start - Missing blank line at beginning of function.
		Line 14: space-before-function-paren - Missing space before function parentheses.
		Line 89: space-before-function-paren - Missing space before function parentheses.
```

File: lib/index.js (hunk 1 above)
Comment (zerok) [inline lib/index.js]: Because they were unused and therefore linting complained :)

File: lib/private.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                handler: function (request, reply) {
+
+                    return reply(Util.format('Hello, %s!', Hoek.escapeHtml(request.auth.credentials.username)));
+                }
+            }
+        });
+
+        return next();
+    });
+
```
Comment (AdriVanHoudt) [inline lib/private.js]: no need for newline here

File: lib/private.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        server.route({
+            method: 'GET',
+            path: '/private',
+            config: {
+                description: 'Returns a welcome message to the authenticated user',
+                auth: 'simple',
+                handler: function (request, reply) {
+
+                    return reply(Util.format('Hello, %s!', Hoek.escapeHtml(request.auth.credentials.username)));
```
Comment (AdriVanHoudt) [inline lib/private.js]: is the username html? why escape it?

File: lib/start.js [hunk 5]
```diff
@@ -4,14 +4,8 @@ var Hoek = require('hoek');
 var Server = require('./index');
 
 
-// Declare internals
-
-var internals = {};
```
Comment (AdriVanHoudt) [inline lib/start.js]: again why remove this? This is here due to the hapi style guide

File: lib/private.js (hunk 4 above)
Comment (zerok) [inline lib/private.js]: Because a username in general should be considered user-provided data and therefore should be escaped.

Comment (AdriVanHoudt) [inline lib/private.js]: Ok I can see that, imo a bit unnecessary here but ok

File: lib/index.js (hunk 1 above)
Comment (AdriVanHoudt) [inline lib/index.js]: are you using the build in lab linter?

**Comment by zerok** [comment]
@AdriVanHoudt This is just plain weird. Seems like eslint is getting mixed into the linting result or something. I was running the linter constantly and got no compains in the end. Will fix them in the evening.

**Comment by AdriVanHoudt** [comment]
@zerok do you have the latest version of Lab? There were some rules added recently

**Comment by zerok** [comment]
@AdriVanHoudt I should have. The issue was that I also had a self-made .eslintrc file lying around which influenced what the linter was doing.

**Comment by AdriVanHoudt** [comment]
Ah that was probably it. you can always see what the test results are on travis to see the results in a 'clean' environment

File: lib/private.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+// Declare internals
+
+var internals = {};
+
+
+internals.validate = function (username, password, callback) {
+
+    var matchingUsers = Users.filter(function (account) {
```
Comment (hueniverse) [inline lib/private.js]: This is far too complicated. Just do a key lookup by username and then check for the password value.
